halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/execute/v1/ExecuteShow.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.execute.v1;
+
+import com.netflix.spinnaker.halyard.cli.command.v1.ShowCommand;
+import com.netflix.spinnaker.halyard.config.model.v1.Halconfig;
+
+public class ExecuteShow extends Executor<ShowCommand> {
+  public ExecuteShow(ShowCommand command) {
+    super(command);
+  }
+
```

Comment:
This seems complicated structurally.
You have a command object that is acting as an Executor factory. The instances it creates are templated back into the command class and all its really adding here is a method that essentially delegates to a lambda expression.

Why not just have it provide the lambda instead of the executor and cut all the intermediary boilerplate classes and code?

Given the diversity of halyard, I wont be surprised if in fact you do need different types of executors. But then at least perhaps have a LambdaExecutor where ShowCommand would instantiate one of those (no need for subclassing) and give it the lambda.

If you take the encapsulation approach I mentioned earlier, then nobody else is using the executor directly anyway so doesnt know the difference. The individual commands become simpler and more concise to write and maintain. And if you do need to change the executor, you can still do that with a local change and consumers are unaffected (plus, you can share those other executors in a similar way if other commands have the same refined needs)